### PR TITLE
test(linter): Add more tests for the `eslint/no-empty-function` rule.

### PR DESCRIPTION
**Pull request by connorshea**
This ports various tests from the original rule, all were added upstream in https://github.com/eslint/eslint/commit/dcd95aafa33a95c8102834af85129f6f398fe394

Most of them pass, but a few needed to be commented out. All of this work was done manually, no Claude allowed because I didn't want to risk hallucinations.

See also #19391, which is about fixing the tests here that fail.

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/no_empty_function.rs [hunk 1]
```diff
@@ -625,6 +637,371 @@ fn test() {
...
+        ),
+        (
+            "const A = class {static *foo(param: string) {}};",
+            Some(serde_json::json!([{ "allow": ["generatorMethods"] }])),
+        ),
+        // TODO: Fix.
+        // (
+        //     "const obj = {get foo(): string {}};",
+        //     Some(serde_json::json!([{ "allow": ["getters"] }])),
+        // ),
```
[the remark this answers is not in the document]
Comment (connorshea) [reply crates/oxc_linter/src/rules/eslint/no_empty_function.rs:740]: yolo!

**Review by camc314 — APPROVED** [review]

**Comment by camc314** [comment]
### Merge activity

* **Feb 14, 10:44 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Feb 14, 10:44 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).
* **Feb 14, 10:50 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

**Comment by camc314** [comment]
* **#19393** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/19393?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19392** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/19392?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/19392?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=19392).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
